components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,357 @@
+  return [MDCActionSheetController actionSheetControllerWithTitle:nil message:nil];
+}
+
+- (instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super initWithNibName:nil bundle:nil];
+  if (self) {
+    _actionSheetTitle = [title copy];
+    _message = [message copy];
+    [self commonMDCActionSheetControllerInit];
+  }
+
+  return self;
+}
+
+- (void)commonMDCActionSheetControllerInit {
+  needsPreferredContentSizeUpdate = false;
+  _transitionController = [[MDCBottomSheetTransitionController alloc] init];
+  _transitionController.dismissOnBackgroundTap = YES;
+  /**
+   We have overriden the self implementation so we don't want others to change the values so we
+  have to call super
+   */
+  super.transitioningDelegate = _transitionController;
+  super.modalPresentationStyle = UIModalPresentationCustom;
+  _dataSource = [[MDCActionSheetDataSource alloc] init];
+  _tableViewController = [[UITableViewController alloc] initWithStyle:UITableViewStylePlain];
+  _tableViewController.tableView.delegate = self;
+  _tableViewController.tableView.translatesAutoresizingMaskIntoConstraints = NO;
+  _tableViewController.tableView.estimatedRowHeight = 56.f;
+  _tableViewController.tableView.rowHeight = UITableViewAutomaticDimension;
+  _tableViewController.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+  _tableViewController.tableView.scrollEnabled = NO;
+  _tableViewController.tableView.dataSource = _dataSource;
+  [_tableViewController.tableView registerClass:[MDCActionSheetItemTableViewCell class]
+                         forCellReuseIdentifier:kReuseIdentifer];
+  self.backgroundColor = [UIColor whiteColor];
+}
+
+- (void)addAction:(MDCActionSheetAction *)action {
+  [_dataSource addAction:action];
+  [_tableViewController.tableView reloadData];
+  [_tableViewController.tableView setNeedsLayout];
+  [self setPreferredContentSizeUpdate];
+}
+
+- (NSArray<MDCActionSheetAction *> *)actions {
+  return _dataSource.actions;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  CGRect headerRect = CGRectZero;
+  headerRect.size.width = CGRectGetWidth(self.view.bounds);
+  _header = [[MDCActionSheetHeaderView alloc] initWithFrame:headerRect];
+  _header.title = _actionSheetTitle;
+  _header.message = _message;
+  _header.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
+
+  [self.view addSubview:_tableViewController.tableView];
+  [self.view addSubview:_header];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  if (needsPreferredContentSizeUpdate == false) {
+    [self layoutViews];
+  }
+}
+
+- (void)layoutViews {
+  CGFloat headerHeight = [_header sizeThatFits:self.view.bounds.size].height;
+  CGFloat width = CGRectGetWidth(self.view.bounds);
+  CGRect tableFrame = _tableViewController.tableView.frame;
+  tableFrame.size.width = width;
+  _tableViewController.tableView.frame = tableFrame;
+  [_tableViewController.tableView reloadData];
```

Comment:
Does the layout calculation work if you don't call reloadData?

Reply:
Removed